src/main/resources/assets/computercraft/lua/rom/apis/paintutils.lua

```diff
@@ -9,25 +9,49 @@ for n=1,16 do
     tColourLookup[ string.byte( "0123456789abcdef",n,n ) ] = 2^(n-1)
 end
 
+local function parseLine( tImageArg, sLine )
+    local tLine = {}
+    for x=1,sLine:len() do
+        tLine[x] = tColourLookup[ string.byte(sLine,x,x) ] or 0
+    end
+    table.insert( tImageArg, tLine )
+    return tImageArg
+end
+
+function parseImage( sRawData )
+    if type( sRawData ) ~= "string" then
+        error ( "bad argument #1 (expected string, got " .. type( sRawData ) .. ")" )
+    end
+    local function split( str, delim ) -- modified split function from https://codea.io/talk/discussion/2118/split-a-string-by-return-newline
+        if string.find( str, delim ) == nil then return { str } end
+        local result, pat, lastpos = {}, "(.-)" .. delim .. "()", nil
+        for part, pos in string.gfind( str, pat ) do table.insert( result, part ); lastpos = pos; end
+        table.insert( result, string.sub( str, lastpos ) )
+        return result
+    end
+    local tImage = {}
+    for _, sLine in pairs( split( sPath, "\n" ) ) do -- read each line like original file handling did
```

Comment:
I wonder if it would be cleaner to iterate using `:gmatch` instead of splitting the string and then looping over it. You also don't need to reassign `tImage` on the next line.